skstore/ts/src/skstore_api.ts

```diff
@@ -0,0 +1,588 @@
+  /**
+   * Get a value of a eager reactive map
+   * @param key - the key of the to get
+   * @returns the value corresponding to the key
+   * @throws {Error} when key not exist
+   */
+  get(key: K): V;
+  /**
+   * Get a value of a eager reactive map if exist
+   * @param key - the key of the to get
+   * @returns the value corresponding to the key if exist else null
+   */
+  maybeGet(key: K): Opt<V>;
+
+  /**
+   * Map over each eager reative map entry and apply mapper function
+   * @param {Mapper} mapper - function to apply to the table entry
+   * @returns {EHandle} The the resulting eager reactive map handle
+   */
+  map<K2 extends TJSON, V2 extends TJSON>(
+    mapper: Mapper<K, V, K2, V2>,
+  ): EHandle<K2, V2>;
+  /**
+   * Map over each eager reative map entry and apply mapper function
+   *  then reduce the when the given accumulator
+   * @param {Mapper} mapper - function to apply to the eager map entry
+   * @param {Accumulator} accumulator - reduction manager
+   * @returns {EHandle} The the resulting eager reactive map handle
+   */
+  mapReduce<K2 extends TJSON, V2 extends TJSON, V3 extends TJSON>(
+    mapper: Mapper<K, V, K2, V2>,
+    accumulator: Accumulator<V2, V3>,
+  ): EHandle<K2, V3>;
+
+  /**
+   * Access the reactive size of the reactive eager map
+   */
+  size: () => number;
+
+  /**
+   * Map over each eager reative map entry to write it into given table
+   * @param {Mapper} mapper - function to apply to the table entry
+   */
+  mapTo<R extends TJSON[]>(
+    table: TableHandle<R>,
+    mapper: OutputMapper<R, K, V>,
+  ): void;
+
+  getId(): string;
+}
+
+export interface TableHandle<R extends TJSON[]> {
+  /**
+   * @returns {string} The name of the table
+   */
+  getName(): string;
+  /**
+   * Lookup in the table using specified index
+   * @param key - the key to lookup in the table
+   * @param index - the index which you want lookup the table
+   * @returns The results of the lookup
+   * @throws {TableIndexError} when an index is not found
+   *          or the index type is not valid
+   */
+  // TODO get(key: TJSON, index?: string): R[];
+
+  /**
+   * Map over each table entry and apply mapper function
+   * @param {EntryMapper} mapper - function to apply to the table entry
+   * @returns {EHandle} The the resulting eager reactive map handle
+   */
+  map<K extends TJSON, V extends TJSON>(
+    mapper: EntryMapper<R, K, V>,
+  ): EHandle<K, V>;
+}
+
+export interface Table<R extends TJSON[]> {
+  getName(): string;
+  /**
+   * Insert an entry in the table
+   * @param entries - the entries to insert in the table
+   * @param update - indicate if the value need to be updated in case of index conflict
+   * @throws {Error} when not update and an index contraints is broken
+   */
+  insert(entry: R[], update?: boolean): void;
+  /**
+   * Update an entry in the table
+   * @param row - the table entry to update
+   * @param updates - the column values updates
```

Comment:
It's just a first quick version of something that not use SQL, because I had the remark during the demo I have done.
That functions are here only to have something running quickly, I did not expect that function to be really used, (except watch and  watchChanges) because the changes shall be performed by the mirroring.
But you are right, if they are here to stay, they need better typing and documentation.